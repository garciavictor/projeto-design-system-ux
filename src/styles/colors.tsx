export const Colors = {
    primary: {
        900: "#E14624",
        700: "#E4593A",
        500: "#E76B50",
        300: "#EA7E66"
    },
    secondary: {
        900: "#084B59",
        700: "#116F84",
        500: "#1A96B1",
        300: "#24BFE1"
    },
    neutrals: {
        white: {
            100: "#F1F0F0",
            75: "#F5F4F4",
            50: "#F7F7F7",
            25: "#FBFBFB"
        },
        black: {
            100: "#1D1A1A",
            75: "#565353",
            50: "#8D8C8C",
            25: "#C6C5C5"
        }
    },
    success: {
        light: "#4ADE80",
        regular: "#22C55E",
        dark: "#16A34A"
    },
    error: {
        light: "#FF4D4D",
        regular: "#FF0000",
        dark: "#CC0000"
    },
    warning: {
        light: "#FACC15",
        regular: "#EAB308",
        dark: "#CA8A04"
    },
    info: {
        light: "#60A5FA",
        regular: "#3B82F6",
        dark: "#2563EB"
    }
}