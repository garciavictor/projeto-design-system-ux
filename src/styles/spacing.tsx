export const Spacing = {
    s1: 4,
    s2: 8,
    s3: 12,
    s4: 16,
    s5: 20,
    s6: 24,
    s7: 32,
    s8: 40,
    s9: 48,
    s10: 64,
}