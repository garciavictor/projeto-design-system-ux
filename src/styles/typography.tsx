export const Typography = {
    h1: {
        weight: '700',
        size: '48px',
        height: '56px'
    },
    h2: {
        weight: '500',
        size: '36px',
        height: '4px'
    },
    h3: {
        weight: '500',
        size: '28px',
        height: '34px'
    },
    text: {
        weight: '400',
        size: '16px',
        height: '24px'
    },
    small: {
        weight: '200',
        size: '14px',
        height: '20px'
    },
    button: {
        weight: '500',
        size: '16px',
        height: '24px'
    },
    input: {
        text: {
            weight: '500',
            size: '16px',
            height: '24px'
        },
        label: {
            weight: '600',
            size: '16px',
            height: '24px'
        },
        description: {
            weight: '300',
            size: '14px',
            height: '20px'
        }
    }
}