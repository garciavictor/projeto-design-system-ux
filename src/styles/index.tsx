import { BorderRadius } from "./border-radius";
import { Colors } from "./colors";
import { Spacing } from "./spacing";
import { Typography } from "./typography";

export const Theme = {
    typography: Typography,
    colors: Colors,
    spacing: Spacing,
    border_radius: BorderRadius,
}
