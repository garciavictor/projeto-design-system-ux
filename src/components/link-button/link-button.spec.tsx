import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { LinkButton } from '.';
import { faCoffee } from '@fortawesome/free-solid-svg-icons';
import { ThemeProvider } from 'styled-components';
import { Theme } from '../../styles';

describe('@components/LinkButton', () => {
    it('matches snapshot without icon', () => {
        const { container } = render(
            <ThemeProvider theme={Theme}>
                <LinkButton linkButtonText="Click me" />
            </ThemeProvider>
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with icon', () => {
        const { container } = render(
            <ThemeProvider theme={Theme}>
                <LinkButton linkButtonText="With Icon" icon={faCoffee} />
            </ThemeProvider>
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
