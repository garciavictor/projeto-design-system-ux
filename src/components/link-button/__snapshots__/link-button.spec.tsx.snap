// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@components/LinkButton > matches snapshot with icon 1`] = `
<a
  class="sc-bXdtCk kYNGgM"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-mug-saucer "
    data-icon="mug-saucer"
    data-prefix="fas"
    focusable="false"
    role="img"
    style="padding-right: 16px;"
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M96 64c0-17.7 14.3-32 32-32l320 0 64 0c70.7 0 128 57.3 128 128s-57.3 128-128 128l-32 0c0 53-43 96-96 96l-192 0c-53 0-96-43-96-96L96 64zM480 224l32 0c35.3 0 64-28.7 64-64s-28.7-64-64-64l-32 0 0 128zM32 416l512 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 480c-17.7 0-32-14.3-32-32s14.3-32 32-32z"
      fill="currentColor"
    />
  </svg>
  With Icon
</a>
`;

exports[`@components/LinkButton > matches snapshot without icon 1`] = `
<a
  class="sc-bXdtCk kYNGgM"
>
  Click me
</a>
`;
